### PR TITLE
front: fix comfort type not updated after rs change

### LINKED ISSUE #9020: The confort type doesn't change when the RS is changed
### What happened?

https://github.com/user-attachments/assets/97e57ebf-61eb-4c84-a496-1dc057a47295

### What did you expect to happen?

The correct confort type should be displayed

### How can we reproduce it (as minimally and precisely as possible)?

1. Go to the **Rolling Stock** tab.
2. Select a rolling stock (RS) with multiple comfort types: **Standard**, and/or **Heating**, and/or **Air Conditioning**.
3. Modify the selected RS and choose a new one with only the **Standard** comfort type.

### On which environments the bug occurs?

Recette (SNCF)

### On which browser the bug occurs?

Google Chrome

### OSRD version (top right corner `Account` button > `Informations`)

a5e9ca2

### PR DESCRIPTION
**Pull request by RomainValls**
closes #9020

### CODE REVIEW

File: front/src/modules/rollingStock/components/RollingStockCard/RollingStockCardButtons.tsx
Comment (SharglutDev) [inline front/src/modules/rollingStock/components/RollingStockCard/RollingStockCardButtons.tsx:1]: Not sure id should be in the dependency array. Also If I open back the rs selector, it switches to standard even though I selected heating for example. I don't think it's what we want.

Here is a suggestion : 

```suggestion
  useEffect(() => {
    if (curvesComfortList.length === 0) {
      setComfort('STANDARD');
    } else {
      setComfort(
        curvesComfortList.includes(currentComfortInStore)
          ? currentComfortInStore
          : curvesComfortList[0]
      );
    }
  }, [curvesComfortList, currentComfortInStore]);
```

Comment (RomainValls) [reply front/src/modules/rollingStock/components/RollingStockCard/RollingStockCardButtons.tsx:1]: Thanks, it's better this way 👍 !

**Review by SharglutDev — APPROVED** [review]
Lgtm and tested, thank you :)

**Review by achrafmohye — APPROVED** [review]
LGTM :)

**Pull request merged**